capstone-rs/src/instruction.rs

```diff
@@ -36,6 +36,19 @@ pub type RegIdInt = u16;
 #[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct RegId(pub RegIdInt);
 
+/// Represents how the register is accessed.
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub enum RegAccessType {
```

Comment:
Add an `impl` block that includes getter methods `is_read()` and `is_write()` so that users don't have to write an in themselves.

Reply:
Actually, I implemented such getters at first, but deleted the `impl` before committing. I think the name `is_read` or `is_write` is misleading: If the instruction both reads and writes to the very same register, should `is_read()` return `true`? Changing the name to `contains_read` or `is_read_only` tells them apart, but the name reads strange.

Reply:
I think if the access is read and write, then both getters should return true. Make sure that the documentation comment is clear.

Maybe a better name is something like is_readable.